materialized_sql_view/model/abstract_materialized_sql_view.py

```diff
@@ -0,0 +1,336 @@
+# -*- coding: utf-8 -*-
+
+import psycopg2
+import logging
+from openerp.osv import osv
+from openerp import SUPERUSER_ID
+from abc import ABCMeta, abstractmethod
+
+logger = logging.getLogger(__name__)
+ABSTRACT_MODEL_NAME = 'abstract.materialized.sql.view'
+
+
+class AbstractMaterializedSqlView(osv.AbstractModel):
+
+    """This class is an abstract model to help developer to
+       create/refresh/update materialized view.
+    """
+    _name = ABSTRACT_MODEL_NAME
+    _description = u"This is an helper class to manage materialized SQL view"
+    _auto = False
+
+    _sql_mat_view_name = ''
+    """The name of the materialized sql view.
+       Must be defined in inherit class (using inherit = [])
+    """
```

Comment:
if the property are mandatory we may use the property decorator to raise a explicit message